server/src/main/java/org/elasticsearch/search/aggregations/metrics/GeoBoundsAggregator.java

```diff
@@ -85,57 +80,20 @@ public LeafBucketCollector getLeafCollector(LeafReaderContext ctx,
         return new LeafBucketCollectorBase(sub, values) {
             @Override
             public void collect(int doc, long bucket) throws IOException {
-                if (bucket >= tops.size()) {
-                    long from = tops.size();
-                    tops = bigArrays.grow(tops, bucket + 1);
-                    tops.fill(from, tops.size(), Double.NEGATIVE_INFINITY);
-                    bottoms = bigArrays.resize(bottoms, tops.size());
-                    bottoms.fill(from, bottoms.size(), Double.POSITIVE_INFINITY);
-                    posLefts = bigArrays.resize(posLefts, tops.size());
-                    posLefts.fill(from, posLefts.size(), Double.POSITIVE_INFINITY);
-                    posRights = bigArrays.resize(posRights, tops.size());
-                    posRights.fill(from, posRights.size(), Double.NEGATIVE_INFINITY);
-                    negLefts = bigArrays.resize(negLefts, tops.size());
-                    negLefts.fill(from, negLefts.size(), Double.POSITIVE_INFINITY);
-                    negRights = bigArrays.resize(negRights, tops.size());
-                    negRights.fill(from, negRights.size(), Double.NEGATIVE_INFINITY);
+                if (bucket >= extents.size()) {
+                    long from = extents.size();
+                    extents = bigArrays.grow(extents, bucket + 1);
+                    for (long i = from; i < extents.size(); i++) {
+                        extents.set(i, createGeoExtentAndAccount());
```

Comment:
So this will probably complicate things, but perhaps we should pre-calculate the size that will be added to the breaker and do one batch addition?  

The java docs for `addRequestCircuitBreakerBytes()` mention that it should be batched for performance reasons.  I'm not positive of the reasons, but I do know that if the real-memory breaker is enabled it will also check that and that carries a certain overhead too.